src/main/java/org/embulk/util/aws/credentials/AwsCredentials.java

```diff
@@ -218,4 +274,5 @@ private static InstanceProfileCredentialsProvider createInstanceProfileCredentia
     }
 
     private static final Logger log = LoggerFactory.getLogger(AwsCredentials.class);
+    private static final String ARN_PATTERN = "arn:aws:iam::%s:role/%s";
```

Comment:
Per requirement in https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html for `partition` could you kindy create a hidden configuration like below and used it as a transferable parameter in this `ARN_PATTERN`

```
    @Override
    @Config("aws_partition")
    @ConfigDefault("\"aws\"")
    String getAwsPartition();
```

So in the future if we need to support `aws-cn` and `aws-us-gov`, it would be easier to adapt